### PR TITLE
ocamlPackages.stdint: 0.5.0 -> 0.5.1

### PR DESCRIPTION
**Pull request by rixed**
###### Motivation for this change

I need the bugfixes present in 0.5.1

###### Things done

Notice that the build system has (unfortunately) been changed from makefile to jbuilder which is a
builder that's been deprecated already. Still, given the right command line dune (its successor)
manage to build the thing, if not without some warnings.
Also, the fstar compiler, that uses stdint, does not built ; but it's already broken in master (with the same error message).

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review `~~wip~~` rev HEAD"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/ocaml-modules/zmq/default.nix [hunk 1]
```diff
@@ -18,9 +18,10 @@ stdenv.mkDerivation rec {
     sha256 = "1f5l4bw78y4drabhyvmpj3z8k30bill33ca7bzhr02m55yf6gqpf";
   };
 
-  patches = [
-    ./ocaml-zmq-issue43.patch
-  ];
+  patches =
+    if builtins.compareVersions stdint.version "0.5.1" == -1 then
+      [ ./ocaml-zmq-issue43.patch ]
+    else [];
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/zmq/default.nix]: Please remove the patch, as it is no longer needed.

**Comment by rixed** [comment]
Rebased.
Note that zmq does not build in sandbox mode any longer on master.

File: pkgs/development/ocaml-modules/zmq/default.nix (hunk 1 above)
Comment (vbgl) [reply pkgs/development/ocaml-modules/zmq/default.nix]: The patch file should be removed too.

Comment (rixed) [reply pkgs/development/ocaml-modules/zmq/default.nix]: Oh well, sorry that I need to be micromanaged like that :)

**Comment by vbgl** [comment]
@GrahamcOfBorg build ocamlPackages.zmq-lwt fstar

**Pull request merged**